server/consumer.go

```diff
@@ -745,6 +745,47 @@ func (o *consumer) setLeader(isLeader bool) {
 	}
 }
 
+func (o *consumer) registerNotification(s *Server, acc *Account, subject string) {
+	o.registerOrClearNotification(s, acc, subject, true)
+}
+
+func (o *consumer) clearNotification(s *Server, acc *Account, subject string) {
+	o.registerOrClearNotification(s, acc, subject, false)
+}
+
+func (o *consumer) registerOrClearNotification(s *Server, acc *Account, subject string, register bool) {
+	if register {
+		acc.sl.RegisterNotification(subject, o.inch)
+	} else {
+		acc.sl.ClearNotification(subject, o.inch)
+	}
+	if s == nil {
+		return
+	}
+	gw := s.gateway
+	if !gw.enabled {
+		return
+	}
+	gw.RLock()
+	defer gw.RUnlock()
+	for _, c := range gw.outo {
+		c.mu.Lock()
+		if ei, ok := c.gw.outsim.Load(acc.Name); ok && ei != nil {
```

Comment:
What about changes after this call?

Reply:
Anyway, I realize that registering on both acc.sl and then the gw.sl is bad because for local interest then the register on gateway would produce a "no interest" in the o.inch channel, which would make delivery stop. My test did not catch that because there was a single message to sync.
Closing this PR for now, need to think of a different way.